kube/src/oauth2/mod.rs

```diff
@@ -1,248 +1,76 @@
-use std::{
-    env,
-    fs::File,
-    path::PathBuf,
-    time::{Duration, SystemTime, UNIX_EPOCH},
+use std::{env, path::PathBuf};
+
+use tame_oauth::{
+    gcp::{ServiceAccountAccess, ServiceAccountInfo, TokenOrRequest},
+    Error as OauthError, Token,
```

Comment:
The oauth module seems to be gcp only, but that's possibly all we supported before? At least that's what we had special support lying around for. Do you know what the tradeoff is there? It's a thing we've not really tested well before and exists entirely because of external contributors.

Reply:
Yeah, only GCP was supported.

> Do you know what the tradeoff is there?

The obvious drawback is more dependencies for users not needing this. We can add a feature.

As far as I can tell, the original implementation was only retrieving the token if it was expired when creating a `Config`. So, it can send an expired token on request after a while. The replacement does the same at the moment, but we should be able to improve this much easier by using this crate because it has internal cache and returns a valid token when it's not expired.

I found this crate being used here for Kubernetes: https://github.com/EmbarkStudios/buildkite-jobify/blob/32ca44fc181d4e52986372d7e83676bddd1e98fb/src/k8s/config/mod.rs#L94-L138. (I was looking at their projects because I saw their name on [tower-http](https://github.com/tower-rs/tower-http) README).

Reply:
I moved `oauth` to `config::auth` and made it `oauth` opt-in.